test/integration/eni-subnet-selection/eni_subnet_selection_test.go

```diff
@@ -0,0 +1,150 @@
+				Command([]string{"sleep"}).
+				Args([]string{"3600"}).
+				Build()
+
+			deploymentBuilder := manifest.NewBusyBoxDeploymentBuilder(f.Options.TestImageRegistry).
+				Container(container).
+				Replicas(replicaCount).
+				PodLabel(podLabelKey, podLabelVal).
+				NodeSelector("topology.kubernetes.io/zone", *primaryInstance.Placement.AvailabilityZone).
+				Build()
+
+			deployment, err = f.K8sResourceManagers.DeploymentManager().
+				CreateAndWaitTillDeploymentIsReady(deploymentBuilder, utils.DefaultDeploymentReadyTimeout)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Wait for deployment to settle, as if any pods restart, their pod IP will change between
+			// the GET and the validation.
+			time.Sleep(5 * time.Second)
+
+			podList, err = f.K8sResourceManagers.PodManager().
+				GetPodsWithLabelSelector(podLabelKey, podLabelVal)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		JustAfterEach(func() {
+			By("deleting deployment")
+			err := f.K8sResourceManagers.DeploymentManager().DeleteAndWaitTillDeploymentIsDeleted(deployment)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("sleeping to allow CNI Plugin to delete unused ENIs")
+			time.Sleep(time.Second * 90)
+		})
+
+		Context("when using a tagged subnet with /18", func() {
+			BeforeEach(func() {
+				ipv4Limit, err = vpc.GetIPv4Limit(*primaryInstance.InstanceType)
+				Expect(err).ToNot(HaveOccurred())
+
+				replicaCount = (ipv4Limit - 1) * 2
+
+				corednsPods, err := f.K8sResourceManagers.PodManager().
+					GetPodsWithLabelSelector("eks.amazonaws.com/component", "coredns")
+				Expect(err).ToNot(HaveOccurred())
+
+				countCoreDNS = 0
+				for _, pod := range corednsPods.Items {
+					if pod.Status.HostIP == *primaryInstance.PrivateIpAddress {
+						countCoreDNS += 1
+					}
+				}
+
+				// Coredns pods doesn't exist on node: ipv4Limit-1 pods with ips from 192.168.0.0/16
+				// 								       ipv4Limit-1 pods with ips from 100.64.0.0/16
+				// Coredns pods exist on node: ipv4Limit-1-countCoreDNS pods with ips from 192.168.0.0/16
+				//							  ipv4Limit-1+countCoreDNS pods with ips from 100.64.0.0/16
+				expectedCount = ipv4Limit - 1 + countCoreDNS
+			})
+			It(fmt.Sprintf("should have ip addresses in CIDR range %s", cidrRangeString), func() {
+				CheckNumberOfIps(expectedCount, podList)
+			})
+		})
+
+		Context("when using an untagged subnet with /18", func() {
+			BeforeEach(func() {
+				By("Untagging the subnet")
+				_, err = f.CloudServices.EC2().
+					DeleteTags(
+						[]string{createdSubnet},
+						[]*ec2.Tag{
+							{
+								Key:   aws.String("kubernetes.io/role/cni"),
+								Value: aws.String("1"),
+							},
+						},
+					)
+				Expect(err).ToNot(HaveOccurred())
+				expectedCount = 0
+			})
+			It(fmt.Sprintf("should not have ip addresses in CIDR range %s", cidrRangeString), func() {
+				CheckNumberOfIps(expectedCount, podList)
+			})
+		})
+	})
+})
+
+func CheckNumberOfIps(expectedCount int, podList coreV1.PodList) {
```

Comment:
nit: instead of counting IPs, we can just make an EC2 API call to get ENIs attached to the instance and validate their subnets. That's the specific check we want for this test suite